tools/generators/ethlike/contract.go.tmpl

```diff
@@ -44,6 +45,7 @@ type {{.Class}} struct {
 
 func New{{.Class}}(
     contractAddress common.Address,
+    chainId *big.Int,
```

Comment:
We don't do it for the hotfix but I think here we should add validation for `chainId` to error if it is null. EIP155 signer does not do the validation, and if chain ID is nil, the transaction is rejected as not properly signed but without any information that the lack of chain ID is the case.

Reply:
Note that the original `go-ethereum` code is performing this check before constructing EIP155 signer: https://github.com/ethereum/go-ethereum/commit/ed0670cb17a96aafeb9eaaeb9765a42fb6bb5663#diff-18aeac66ddb948fefafc8c96289984cd90c94c54d0616e96165ffff8865b6250R143

Reply:
Done in https://github.com/keep-network/keep-common/pull/72/commits/94d25202ee9d1a1a6c66e6df72b24f49ba6b046a.